packages/expo-splash-screen/plugin/src/withSplashScreen.ts

```diff
@@ -1,14 +1,28 @@
+import { AndroidSplashConfig } from '@expo/prebuild-config/build/plugins/unversioned/expo-splash-screen/getAndroidSplashConfig';
+import { IOSSplashConfig } from '@expo/prebuild-config/build/plugins/unversioned/expo-splash-screen/getIosSplashConfig';
 import { withAndroidSplashScreen } from '@expo/prebuild-config/build/plugins/unversioned/expo-splash-screen/withAndroidSplashScreen';
 import { withIosSplashScreen } from '@expo/prebuild-config/build/plugins/unversioned/expo-splash-screen/withIosSplashScreen';
+import Debug from 'debug';
 import { ConfigPlugin, createRunOncePlugin } from 'expo/config-plugins';
 
+const debug = Debug('expo:prebuild-config:expo-splash-screen');
 const pkg = require('expo-splash-screen/package.json');
 
-const withSplashScreen: ConfigPlugin = (config) => {
+type SplashConfig =
+  | {
+      ios?: IOSSplashConfig;
+      android?: AndroidSplashConfig;
+    }
+  | undefined
+  | null
+  | void;
+
+const withSplashScreen: ConfigPlugin<SplashConfig> = (config, splash = undefined) => {
```

Comment:
```suggestion
const withSplashScreen: ConfigPlugin<SplashConfig> = (config, splash) => {
```

nit: should be implicitly undefined by default

Reply:
didn't see the usage for custom splash for `withSplashScreen` in this pr.
supposedly we will pass the value from app.json's `.expo.plugins[] | select(type == "array" and .[0] == "expo-splash-screen")` to this config-plugin?